### DISCUSSION TITLE
Add device classes to lights

### DISCUSSION THREAD

**Discussion by emontnemery** [home-assistant/architecture#1015] [Entity Models] [answered]
## Background

Light entities are used to model a wide range of things, just to name a few examples:
- Room lighting
- Decorative light
- Indicator light
- Screen backlight
- Security camera IR illuminator

It's unlikely that a user request to "turn off the lights in the bedroom" means to turn off all of these types. This problem is partially solved by entity categories; the indicator light in the example above is likely to be categorized as a configuration entity.

## Proposal

Add device classes to lights, for example:
- A class for decorative lights
- A catch-all class for lights which are not light fixtures or lamps (to cover examples such as indicator lights and security camera IR illuminator examples above)
- Maybe classes by the function of a light: ambient, task and accent lighting
- Maybe classes by the type of light fixture or lamp: ceiling light, desk lamp

It should be possible for the user to override the device class of a light from the UI.

**Comment by Petro31** [comment]
There are so many that would be nice to have.

* Recessed
* Decorative
* Spotlight / FloodLight
* Indicator
* Screen Backlight
* Lamp
* String
* Bulb (default)
* Chandelier

**Reply by tetele** [reply]
I think that's a mixture of types of lights and purposes of lights and these should be separated. I agree we should assign `device_class`es based on purpose/destination, but not on form factor. That's a different taxonomy

**Reply by EnochPrime** [reply]
> that's a mixture of types of lights and purposes of lights and these should be separated

I thought this at first, but this is similar to the cover entity device classes. Blind, curtain, shade all have the same purpose of blocking light. Meanwhile you can also select garage door, gate, or window.

**Reply by Petro31** [reply]
The problem is device_class is specifically used for form factor in many other places.  Roughly to just handle translations and icons.  If this has been expanded on to be used in other areas of HA, then there should be no reason to avoid the granularity here.

**Reply by ScottG489** [reply]
Isn't the risk here that there could be overlap? You can have a "string" light that's also "decorative".

**Reply by tetele** [reply]
1. There's no need to dig ourselves deeper in the hole we're already in
2. I believe the purpose @emontnemery set out with was to discern what should be exposed/toggleable and what not, and IMHO light purpose defines that, not form factor.

**Reply by Petro31** [reply]
Overlap isn't really a problem.  Grouping device_class checks in the backend is just another if statement.

**Reply by Petro31** [reply]
> 1. There's no need to dig ourselves deeper in the hole we're already in
> 2. I believe the purpose @emontnemery set out with was to discern what should be exposed/toggleable and what not, and IMHO light purpose defines that, not form factor.

We are already in the hole wtih other domains.  They have to be accounted for anyways.  A system has to be put in place to account for those, there's no reason that this granularity has to be avoided other than a personal opinion.

**Reply by tetele** [reply]
My proposal is not lack of granularity, but taxonomy and separation of concerns

**Reply by Petro31** [reply]
And if we put in half a feature that disrupts the look and feel of existing lights, it will just lead to an upset group of users.  Especially if we don't have anything for them to adjust.  We should not add half a feature just to address voice concerns when it impacts peoples user interfaces.  We've done this in the past and it always leads towards outrage.  Device_class is locked to a look and feel of the lights in the user interface.

**Reply by EnochPrime** [reply]
I'm wondering now if the difference of opinion here works itself out with #1017. If the proposed `area_responsibility` is added you could have a the `device_class` be string and the `area_responsibility` be decorative.

**Reply by emontnemery** [reply]
Based on the discussion, I suggest to limit the device classes to:
- A class for lights which provide room or outdoor lighting
- A class for decorative lights
- A catch-all class for lights which are not light fixtures or lamps (to cover examples such as indicator lights and security camera IR illuminator examples above)

Are there any suggestions for names of these classes?

@Petro31 To address the concern with UI look and feel, I suggest to make light device classes configurable from the UI. Icon is already configurable.

**Reply by gjohansson-ST** [reply]
- Ambient
- Decorative
- Other (or should the catch-all perhaps be to not have a device class like we mostly do otherwise?)

**Reply by EnochPrime** [reply]
This is more than 3, but some external sources already have a defined set of light types which seem like a good starting point. (https://richporterlighting.com/en/blog-what-are-the-different-types-of-lighting/)
- Ambient
- Task
- Accent
- Decorative

**Comment by frenck** [comment, marked as answer]
We have discussed this topic multiple times over the years, but never concluded. The general agreement for now is that assigning device classes is hard, as it really depends on the user's use case. The provided lists above are probably not even exhausting at this point.

We might pick up again by the time we implement the use of actual device types in Home Assistant (which we plan to start working on later this year).

For now, I'll go ahead and close this proposal.

../Frenck
